### PR TITLE
Allow "friendsofsymfony/rest-bundle:^3.0"

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

Allow "friendsofsymfony/rest-bundle:^3.0".

I am targeting this branch, because these changes respect BC.

## Changelog

```markdown
### Removed
- Removed support for deprecated "rest" routing type
```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -40,21 +40,24 @@
...
     },
     "require-dev": {
         "enqueue/amqp-lib": "^0.8",
-        "friendsofsymfony/rest-bundle": "^2.1",
+        "friendsofsymfony/rest-bundle": "^2.6 || ^3.0",
         "guzzlehttp/guzzle": "^3.8",
         "jms/serializer-bundle": "^2.0 || ^3.0",
         "liip/monitor-bundle": "^2.0",
         "nelmio/api-doc-bundle": "^2.4",
+        "sensio/framework-extra-bundle": "^5.5",
```
Comment (jordisala1991) [inline composer.json:55]: Why is this needed?

Comment (wbloszyk) [reply composer.json:55]: For smoke tests, oterwise `Error: Class 'Sensio\Bundle\FrameworkExtraBundle\Configuration\Template' not found`

**Comment by wbloszyk** [comment]
I bump some package to version from `composer update --prefer-lowest`, result:
- [before: 10 min 16 sec](https://travis-ci.org/github/sonata-project/SonataNotificationBundle/builds/711389975)
- [after:  4 min 6 sec](https://travis-ci.org/github/sonata-project/SonataNotificationBundle/builds/711394598)

File: composer.json [hunk 2]
```diff
@@ -23,38 +23,41 @@
...
         "doctrine/doctrine-bundle": "^1.12 || ^2.0",
-        "doctrine/persistence": "^1.3",
+        "doctrine/persistence": "^1.3.4",
         "sonata-project/datagrid-bundle": "^2.5",
         "sonata-project/doctrine-extensions": "^1.5",
         "sonata-project/form-extensions": "^0.1 || ^1.4",
         "symfony/config": "^4.4",
         "symfony/console": "^4.4",
-        "symfony/dependency-injection": "^4.4",
+        "symfony/dependency-injection": "^4.4.3",
```
Comment (jordisala1991) [inline composer.json:32]: I would avoid this change if not completely necessary, to make consistent will all bundles requiring just ^4.4

Comment (wbloszyk) [reply composer.json]: > I bump some package to version from `composer update --prefer-lowest`, result:
> 
> * [before: 10 min 16 sec](https://travis-ci.org/github/sonata-project/SonataNotificationBundle/builds/711389975)
> * [after:  4 min 6 sec](https://travis-ci.org/github/sonata-project/SonataNotificationBundle/builds/711394598)

I want avoid timeout for ``--prerfer-lowest` tests.  Lets check it for composer without patch version: https://travis-ci.org/github/sonata-project/SonataNotificationBundle/builds/711399641

Comment (wbloszyk) [reply composer.json]: 4 min 17 sec :smiley:

Comment (jordisala1991) [reply composer.json]: this concrete changes makes all the differente? 🤔

**Comment by wbloszyk** [comment]
RTM. @phansys Can you check it?

File: tests/App/config/routes.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#NelmioApiDocBundle:
```
Comment (phansys) [inline tests/App/config/routes.yaml:1]: Why is this comment required?

File: tests/Functional/RoutingTest.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        $route = $router->getRouteCollection()->get($name);
+
+        $this->assertNotNull($route);
+        $this->assertSame($path, $route->getPath());
+        $this->assertEmpty(array_diff($methods, $route->getMethods()));
+    }
+
+    public function getRoutes(): iterable
+    {
+        //yield ['nelmio_api_doc_index', '/api/doc/{view}', ['GET']];
```
[the remark this answers is not in the document]
Comment (wbloszyk) [reply tests/Functional/RoutingTest.php]: Fix

File: tests/App/config/routes.yaml (hunk 3 above)
Comment (wbloszyk) [reply tests/App/config/routes.yaml]: Fix

**Comment by phansys** [comment]
Thank you @wbloszyk!
